sale_order_line_delivery_state/models/sale_order_line.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2021 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools import float_compare, float_is_zero
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    delivery_state = fields.Selection(
+        [
+            ("no", "No delivery"),
+            ("unprocessed", "Unprocessed"),
+            ("partially", "Partially processed"),
+            ("done", "Done"),
+        ],
+        # Compute method have a different name then the field because
+        # the method _compute_delivery_state already exist to compute
+        # the field delivery_set in odoo delivery module
```

Comment:
In Odoo 17, this no longer seems to be the case. There is a `_compute_delivery_state` method, still in the Odoo delivery module, but it's only on `sale.order`, not on `sale.order.line`.